modelhub/modelhub/modelhub.py

```diff
@@ -298,3 +298,74 @@ def get_funnel_discovery(self) -> FunnelDiscovery:
         """
 
         return FunnelDiscovery()
+
+    def visualize_location_stack(self,
+                                 data: bach.DataFrame,
+                                 root_location: str = None,
+                                 location_stack: 'SeriesLocationStack' = None,
+                                 n_top_examples=40,
+                                 return_df: bool = False):
+        """
+        Shows the location stack as a sankey chart per element for the selected root location. It shows the
+        different elements by type and id as nodes from left to right. The size of the nodes and links
+        indicate the number of events that have this location element or these location element combinations.
+
+        :param data: :py:class:`bach.DataFrame` to apply the method on.
+        :param root_location: the name of the root location to use for visualization of the location stack.
+            If None, it will use the most common root location in the data.
+        :param location_stack: if None, it will use the standard location stack.
+        :param n_top_examples: number of top examples  from the location stack to plot (if we have
+            too many examples to plot it can slow down the browser).
+        :param return_df: returns a :py:class:`bach.DataFrame` with the data from which the sankey diagram is
+            created.
+        :returns: None or DataFrame
+        """
+
+        from modelhub.util import check_objectiv_dataframe
+
+        if location_stack is None:
+            check_objectiv_dataframe(df=data, columns_to_check=['location_stack', 'user_id', 'event_id'])
+
+        data_cp = data.copy()
+        result_item, result_offset = data_cp.location_stack.json.flatten_array()
```

Comment:
```suggestion
        if location_stack is None:
            check_objectiv_dataframe(df=data, columns_to_check=['location_stack', 'user_id', 'event_id'])
            location_stack = data.location_stack

        data_cp = data.copy()
        result_item, result_offset = location_stack.json.flatten_array()
```

Otherwise `location_stack` parameter is not used.

Reply:
ah good point. maybe I will adjust this.